### ISSUE TITLE
IndexError when predicting one sample

### ISSUE THREAD

**Issue by Visdoom** [civisanalytics/python-glmnet#30]
Hey there, 

when doing leave one out cross-validation `LogitNet` fails with an `Index Error` when I try to predict the label of the single test sample.
Here is a simple working example:
```
X = np.random.randint(0,10, size=(22,10))
y = np.random.randint(0,2,size=(22,))


X_train = X[:-1,:]
y_train = y[:-1]

X_test = X[[-1]]
y_test = y[[-1]]

m = LogitNet(alpha=0.8, tol=0.3, max_iter=2000)

m.fit(X_train, y_train)

m.predict(X_test)
```

In my case it fails with
```
---------------------------------------------------------------------------
IndexError                                Traceback (most recent call last)
<ipython-input-178-2b6b26202b08> in <module>()
     15 m.fit(X_train, y_train)
     16 
---> 17 m.predict(X_test)

/usr/local/lib/python3.5/dist-packages/glmnet/logistic.py in predict(self, X, lamb)
    474         """
    475 
--> 476         scores = self.predict_proba(X, lamb)
    477         indices = scores.argmax(axis=1)
    478 

/usr/local/lib/python3.5/dist-packages/glmnet/logistic.py in predict_proba(self, X, lamb)
    443         # reshape z to (n_samples, n_classes, n_lambda)
    444         n_lambda = len(np.atleast_1d(lamb))
--> 445         z = z.reshape(z.shape[0], -1, n_lambda)
    446 
    447         if z.shape[1] == 1:

IndexError: tuple index out of range

```

As a side note, I could imagine it refers to #25 but I am not sure. 

When switching to leave-2-out cv the problem does not occur anymore.
Best and Merry Christmas!

**Comment by xdavio** [comment]
Hey, thanks for catching this! I think the fix will be changing line 445 (in your error log) to
```python
z = np.atleast_1d(z)
z = z.reshape(z.shape[0], -1, n_lambda)
```

I'll poke around and introduce a fix later.

**Comment by stephen-hoover** [comment]
I also ran into this. The `glment.ElasticNet.predict` method outputs an array with dimension 0 when given one row to predict. It should return an array with dimension 1 and shape (1,). Presumably the `LogitNet.predict_proba` has the same problem.

**Comment by stephen-hoover** [comment]
I think this issue was the same thing as #25 , and it seems to have been fixed by #51 . My above report is not yet fixed. I've filed it as issue #52 .
